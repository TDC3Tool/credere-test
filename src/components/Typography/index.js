import React from 'react'
import PropTypes from 'prop-types'
import {
  Heading, Paragraph, SubHeading, Span,
} from './styles'

const Typography = ({
  tag, children, color,
  emphasys, ...rest
}) => {
  switch (tag) {
    case 'span':
      return (
        <Span style={{ color }} emphasys={emphasys} {...rest}>
          {children}
        </Span>
      )

    case 'heading':
      return (
        <Heading style={{ color }} emphasys={emphasys} {...rest}>
          {children}
        </Heading>
      )

    case 'sub-heading':
      return (
        <SubHeading style={{ color }} emphasys={emphasys} {...rest}>
          {children}
        </SubHeading>
      )

    case 'paragraph':
      return (
        <Paragraph style={{ color }} emphasys={emphasys} {...rest}>
          {children}
        </Paragraph>
      )

    default:
      return (
        <Span style={{ color }} emphasys={emphasys} {...rest}>
          {children}
        </Span>
      )
  }
}

Typography.propTypes = {
  children: PropTypes.node.isRequired,
  tag: PropTypes.oneOf(['heading', 'sub-heading', 'paragraph', 'span', '']),
  color: PropTypes.string,
  emphasys: PropTypes.bool,
}

Typography.defaultProps = {
  emphasys: false,
  tag: '',
  color: '#FFFFFF',
}

Typography.Span = Span
Typography.Heading = Heading
Typography.SubHeading = SubHeading
Typography.Paragraph = Paragraph

export default Typography
