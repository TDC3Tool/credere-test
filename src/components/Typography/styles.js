import styled from 'styled-components'
import { setFontSizing, isEmphasys } from '../../utils/font'
import { mobileSm } from '../../globalStyles'

export const Heading = styled.h2`
  ${props => isEmphasys(props.emphasys)};
  text-transform: uppercase;
  ${setFontSizing(32)}

  @media screen and (max-width: ${mobileSm}) {
    ${setFontSizing(22)}
  }
`
export const SubHeading = styled.h3`
  ${props => isEmphasys(props.emphasys) || 'font-weight: 400'};
  ${setFontSizing(22)}

  @media screen and (max-width: ${mobileSm}) {
    ${setFontSizing(18)}
  }
`
export const Paragraph = styled.p`
  ${props => isEmphasys(props.emphasys)};
  ${setFontSizing(18)}

  @media screen and (max-width: ${mobileSm}) {
    ${setFontSizing(14)}
  }
`

export const Span = styled.span``
