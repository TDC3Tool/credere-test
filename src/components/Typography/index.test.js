import React from 'react'
import renderer from 'react-test-renderer'
import Typography from '.'

describe('Typography Component', () => {
  it('should render compound typography components', () => {
    const tree = renderer.create(
      <>
        <Typography.Span>Span</Typography.Span>
        <Typography.Paragraph>SubHeading</Typography.Paragraph>
        <Typography.SubHeading>SubHeading</Typography.SubHeading>
        <Typography.Heading>Heading</Typography.Heading>
      </>,
    ).toJSON()

    expect(tree).toMatchSnapshot()
  })

  it('should render Typography Span with props', () => {
    const props = {
      text: 'Test Span',
      color: '#333',
      type: 'span',
    }
    const element = renderer.create(
      <Typography color={props.color} tag="span" emphasys>{props.text}</Typography>,
    ).toJSON()

    expect(element).toMatchObject({
      type: props.type,
      children: [props.text],
      props: {
        style: {
          color: props.color,
        },
      },
    })

    expect(element).toMatchSnapshot()
  })

  it('should render Typography Heading with props', () => {
    const props = {
      text: 'Test Heading',
      color: '#333',
      type: 'h2',
    }

    const element = renderer.create(
      <Typography color={props.color} tag="heading" emphasys>{props.text}</Typography>,
    ).toJSON()

    expect(element).toMatchObject({
      type: props.type,
      children: [props.text],
      props: {
        style: {
          color: props.color,
        },
      },
    })

    expect(element).toMatchSnapshot()
  })

  it('should render Typography sub-heading with props', () => {
    const props = {
      text: 'Test Sub Heading',
      color: '#ccc',
      type: 'h3',
    }

    const element = renderer.create(
      <Typography color={props.color} tag="sub-heading" emphasys>{props.text}</Typography>,
    ).toJSON()

    expect(element).toMatchObject({
      type: props.type,
      children: [props.text],
      props: {
        style: {
          color: props.color,
        },
      },
    })

    expect(element).toMatchSnapshot()
  })

  it('should render Typography paragraph with props', () => {
    const props = {
      text: 'Test Paragraph',
      color: '#ccc',
      type: 'p',
    }

    const element = renderer.create(
      <Typography color={props.color} tag="paragraph" emphasys>{props.text}</Typography>,
    ).toJSON()

    expect(element).toMatchObject({
      type: props.type,
      children: [props.text],
      props: {
        style: {
          color: props.color,
        },
      },
    })

    expect(element).toMatchSnapshot()
  })
})
