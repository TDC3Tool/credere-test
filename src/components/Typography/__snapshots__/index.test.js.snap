// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Typography Component should render Typography Heading with props 1`] = `
<h2
  className="sc-bdVaJa fpQCmI"
  style={
    Object {
      "color": "#333",
    }
  }
>
  Test Heading
</h2>
`;

exports[`Typography Component should render Typography Span with props 1`] = `
<span
  className="sc-bxivhb bUZGRr"
  style={
    Object {
      "color": "#333",
    }
  }
>
  Test Span
</span>
`;

exports[`Typography Component should render Typography paragraph with props 1`] = `
<p
  className="sc-htpNat cmjWi"
  style={
    Object {
      "color": "#ccc",
    }
  }
>
  Test Paragraph
</p>
`;

exports[`Typography Component should render Typography sub-heading with props 1`] = `
<h3
  className="sc-bwzfXH bJNhon"
  style={
    Object {
      "color": "#ccc",
    }
  }
>
  Test Sub Heading
</h3>
`;

exports[`Typography Component should render compound typography components 1`] = `
Array [
  <span
    className="sc-bxivhb bUZGRr"
  >
    Span
  </span>,
  <p
    className="sc-htpNat cSPdLd"
  >
    SubHeading
  </p>,
  <h3
    className="sc-bwzfXH jmxchW"
  >
    SubHeading
  </h3>,
  <h2
    className="sc-bdVaJa bqTftu"
  >
    Heading
  </h2>,
]
`;
